### ISSUE TITLE
Crop portion of mp4

### ISSUE THREAD

**Issue by FunnyDevs** [MasayukiSuda/Mp4Composer-android#36]
Is possible?

**Comment by MasayukiSuda** [comment]
yes. use FillModeCustomItem

**Comment by FunnyDevs** [comment]
I'm trying to crop a square portion of 300X300 pixel in a 720x1280 video with offset in Y. I use

```
                .size(300, 300)
                .fillMode(FillMode.CUSTOM)
                .customFillMode(new FillModeCustomItem(1,0,0,200,300,300))
```

But i have a black video as a result

**Comment by MasayukiSuda** [comment]
```
.size(300, 300)
.fillMode(FillMode.PRESERVE_ASPECT_CROP)
```
try it

**Comment by FunnyDevs** [comment]
With this code it crops from center. I would like to start from top=0, left=0 and apply the crop portion moving from it with offsetY and offsetX. 
FFmpeg use "crop=out_w:out_h:x:y" but i don't want to use ffmpeg.

**Comment by FunnyDevs** [comment]
any news?

**Comment by abellee** [comment]
@MasayukiSuda seems only can crop the center of the video right?

**Comment by MasayukiSuda** [comment]
@FunnyDevs @abellee 
I am sorry for being late. Take a look at the FillModeCustom sample we created.
Please check the behavior as well.

https://github.com/MasayukiSuda/Mp4Composer-android/blob/master/sample/src/main/java/com/daasuu/sample/FillModeCustomActivity.java#L155-L162

**Comment by abellee** [comment]
> @FunnyDevs @abellee
> I am sorry for being late. Take a look at the FillModeCustom sample we created.
> Please check the behavior as well.
> 
> https://github.com/MasayukiSuda/Mp4Composer-android/blob/master/sample/src/main/java/com/daasuu/sample/FillModeCustomActivity.java#L155-L162

so what is this mean?

what the `translateY` exactly is?

why you do this `playerTextureView.getTranslationY() / (baseWidthSize / 9f * 16) * 2f` ?

if I do record the 1280*720 mp4, then I wanna clip the 16:9 size with the 300 translateY. how to do it?

**Comment by fahimfarhan** [comment]
Hello there @abellee and any wonderful person. Recently I was working on a video editor and I had also faced same problem. After some trial and error, I think I have a basic understanding of the crop mechanism. Take a look at my project: [fahimfarhan/SimpleVideoEditor](https://github.com/fahimfarhan/SimpleVideoEditor) I've tried to explain the math and what I understand as simply as possible. I hope it will be helpful :)
